pkg/controller/backupbucket/actuator.go

```diff
@@ -41,7 +41,13 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, backupBucket *e
 		return err
 	}
 
-	azCloudConfiguration, err := azureclient.AzureCloudConfigurationFromCloudConfiguration(backupConfig.CloudConfiguration)
+	cloudConfiguration := backupConfig.CloudConfiguration
+
+	if cloudConfiguration == nil {
+		cloudConfiguration = &azure.CloudConfiguration{Name: helper.CloudInstanceNameFromRegion(backupBucket.Spec.Region)}
+	}
+
+	azCloudConfiguration, err := azureclient.AzureCloudConfigurationFromCloudConfiguration(cloudConfiguration)
```

Comment:
Can't these functions be grouped up in a single function ? Can we imagine using one but not the other ?

Reply:
Unfortunately, there is _one_ case where they are both required. However, I agree that duplicating these lines of code tenfold is not the way to go.

I've tried my hand at refactoring it a bit, but unfortunately there is some inherent ugliness here due to the way the upstream cloud-instance structs are defined. Not sure if I have hit the mark, honestly 🤔